org.adempiere.base/src/org/compiere/acct/Doc_BankStatement.java

```diff
@@ -202,26 +203,52 @@ public ArrayList<Fact> createFacts (MAcctSchema as)
 				// Normal Adempiere behavior -- unchanged if using clearing accounts
 
 				//  BankAsset       DR      CR  (Statement)
-				fl = fact.createLine(line,
-					getAccount(Doc.ACCTTYPE_BankAsset, as),
-					line.getC_Currency_ID(), line.getStmtAmt());
+				fl = fact.createLine(line, acct_bank_asset, line.getC_Currency_ID(), line.getStmtAmt());
 				if (fl != null && AD_Org_ID != 0)
 					fl.setAD_Org_ID(AD_Org_ID);
 				if (fl != null && C_BPartner_ID != 0)
 					fl.setC_BPartner_ID(C_BPartner_ID);
 
-				//  BankInTransit   DR      CR              (Payment)
-				fl = fact.createLine(line,
-					getAccount(Doc.ACCTTYPE_BankInTransit, as),
-					line.getC_Currency_ID(), line.getTrxAmt().negate());
-				if (fl != null)
-				{
-					if (C_BPartner_ID != 0)
-						fl.setC_BPartner_ID(C_BPartner_ID);
-					if (AD_Org_ID != 0)
-						fl.setAD_Org_ID(AD_Org_ID);
-					else
-						fl.setAD_Org_ID(line.getAD_Org_ID(true)); // from payment
+				// BankInTransit DR CR (Payment)
+				MBankStatementLine statementLine = (MBankStatementLine) line.getPO();
+
+				if (statementLine.getC_DepositBatch_ID() != 0) {
+
+					// All Deposit Line
+					MDepositBatchLine[] depositBatchLines = ((MDepositBatch) statementLine.getC_DepositBatch()).getLines();

```

Comment:
More efficient to remove the type cast, override MBankStatementLinet.getC_DepositBatch() to return MDepositBatch (using the MDepositBatch constructor).